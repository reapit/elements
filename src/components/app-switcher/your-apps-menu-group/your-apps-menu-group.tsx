import { AppSwitcherMenuGroupHasAccessContext } from '../menu-group-has-access-context'
import { Menu } from '../../menu'

import type { ReactNode } from 'react'

interface AppSwitcherMenuGroupProps {
  children: ReactNode
}

function AppSwitcherYourAppsMenuGroup({ children }: AppSwitcherMenuGroupProps) {
  return (
    <Menu.Group label={'YOUR APPS'}>
      <AppSwitcherMenuGroupHasAccessContext.Provider value={true}>
        {children}
      </AppSwitcherMenuGroupHasAccessContext.Provider>
    </Menu.Group>
  )
}

export { AppSwitcherYourAppsMenuGroup }
