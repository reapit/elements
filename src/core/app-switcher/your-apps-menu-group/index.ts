export * from './your-apps-menu-group'
