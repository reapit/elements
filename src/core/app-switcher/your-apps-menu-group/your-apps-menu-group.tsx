import { AppSwitcherMenuGroupHasAccessContext } from '../menu-group-has-access-context'
import { DeprecatedMenu } from '#src/deprecated/menu'

import type { ReactNode } from 'react'

interface AppSwitcherMenuGroupProps {
  children: ReactNode
}

function AppSwitcherYourAppsMenuGroup({ children }: AppSwitcherMenuGroupProps) {
  return (
    <DeprecatedMenu.Group label={'YOUR APPS'}>
      <AppSwitcherMenuGroupHasAccessContext.Provider value={true}>
        {children}
      </AppSwitcherMenuGroupHasAccessContext.Provider>
    </DeprecatedMenu.Group>
  )
}

export { AppSwitcherYourAppsMenuGroup }
