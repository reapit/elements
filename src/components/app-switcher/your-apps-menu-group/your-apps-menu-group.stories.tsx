import type { Decorator, Meta, StoryObj } from '@storybook/react'
import { AppSwitcher } from '../app-switcher'
import { AppSwitcherYourAppsMenuGroup } from './your-apps-menu-group'

const useParentDecorator: Decorator = (Story) => {
  return (
    <div style={{ boxSizing: 'content-box', border: '1px solid #FA00FF' }}>
      <Story />
    </div>
  )
}

const meta = {
  title: 'Components/AppSwitcher/YourAppsMenuGroup',
  component: AppSwitcherYourAppsMenuGroup,
  argTypes: {
    children: {
      control: false,
    },
  },
  decorators: [useParentDecorator],
} satisfies Meta<typeof AppSwitcherYourAppsMenuGroup>

export default meta

type Story = StoryObj<typeof meta>

export const Default: Story = {
  args: {
    children: null,
  },
  render: ({}) => {
    return (
      <AppSwitcher.YourAppsMenuGroup>
        <AppSwitcher.ProductMenuItem productId="agentBox" url={globalThis.top?.location.href!} />
      </AppSwitcher.YourAppsMenuGroup>
    )
  },
}
