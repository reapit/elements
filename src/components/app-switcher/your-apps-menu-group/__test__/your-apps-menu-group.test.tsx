import { render } from '@testing-library/react'
import { AppSwitcherYourAppsMenuGroup } from '../your-apps-menu-group'

test('renders AppSwitcherYourAppsMenuGroup properly', () => {
  const { asFragment } = render(<AppSwitcherYourAppsMenuGroup>Fake child</AppSwitcherYourAppsMenuGroup>)
  expect(asFragment()).toMatchSnapshot()
})
