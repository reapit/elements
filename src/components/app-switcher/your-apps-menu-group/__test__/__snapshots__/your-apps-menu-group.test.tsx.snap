// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders AppSwitcherYourAppsMenuGroup properly 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-11 el-menu-item-group"
    role="group"
  >
    <div
      class="mocked-styled-10 el-menu-item-group-title"
    >
      YOUR APPS
    </div>
    <div
      class="mocked-styled-12 el-menu-item-group-list"
      style="max-height: var(undefined);"
    >
      <a
        class="mocked-styled-13 el-app-switcher-menu-item-anchor"
        href="https://www.test.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="mocked-styled-14 el-app-switcher-menu-item-logo"
        >
          <svg
            fill="none"
            font-size="40"
            height="1em"
            viewBox="0 0 40 40"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 8C0 3.58172 3.58172 0 8 0H32C36.4183 0 40 3.58172 40 8V32C40 36.4183 36.4183 40 32 40H8C3.58172 40 0 36.4183 0 32V8Z"
              fill="url(#paint0_linear_203_7948)"
            />
            <g
              clip-path="url(#clip0_203_7948)"
            >
              <path
                d="M8.11963 13.1456V26.8544L19.9909 33.7081L31.8629 26.8544V19.2783L30.2109 18.3242L27.4634 16.738L26.4043 16.1262L26.4029 20.8561L26.4015 25.586H25.1786H14.8018H13.5782V24.3631V17.3757V16.1521L16.7842 14.3013L18.9311 13.0623L19.9902 12.4505L23.1969 14.2887L25.3431 15.5186L26.4043 16.1262V14.9033V9.99423L19.9909 6.29193L8.11963 13.1456Z"
                fill="white"
              />
            </g>
            <defs>
              <lineargradient
                gradientUnits="userSpaceOnUse"
                id="paint0_linear_203_7948"
                x1="20.7019"
                x2="20.7019"
                y1="0.350877"
                y2="31.2281"
              >
                <stop
                  stop-color="#5573FD"
                />
                <stop
                  offset="0.575"
                  stop-color="#4E56EA"
                />
              </lineargradient>
              <clippath
                id="clip0_203_7948"
              >
                <rect
                  fill="white"
                  height="28"
                  transform="translate(6 6)"
                  width="28"
                />
              </clippath>
            </defs>
          </svg>
        </div>
        <div
          class="mocked-styled-15 el-menu-item-text-wrapper"
        >
          <div
            class="mocked-styled-16 el-app-switcher-menu-item-name"
          >
            Reapit PM
          </div>
          <div
            class="mocked-styled-17 el-app-switcher-menu-item-description"
          >
            Property Management
          </div>
        </div>
      </a>
      <a
        class="mocked-styled-13 el-app-switcher-menu-item-anchor"
        href="https://www.test.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="mocked-styled-14 el-app-switcher-menu-item-logo"
        >
          <svg
            fill="none"
            font-size="40"
            height="1em"
            viewBox="0 0 40 40"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 8C0 3.58172 3.58172 0 8 0H32C36.4183 0 40 3.58172 40 8V32C40 36.4183 36.4183 40 32 40H8C3.58172 40 0 36.4183 0 32V8Z"
              fill="url(#paint0_linear_681_72)"
            />
            <g
              clip-path="url(#clip0_681_72)"
            >
              <path
                d="M10.5934 6.28003V9.98303H25.3403V29.1294H29.3891V6.28003H10.5934Z"
                fill="white"
              />
              <path
                d="M20.9394 33.72V13.9604L10.5934 9.98303L10.5934 29.7461L20.9394 33.72Z"
                fill="white"
              />
            </g>
            <defs>
              <lineargradient
                gradientUnits="userSpaceOnUse"
                id="paint0_linear_681_72"
                x1="20.7019"
                x2="20.7019"
                y1="0.350877"
                y2="31.2281"
              >
                <stop
                  stop-color="#5573FD"
                />
                <stop
                  offset="0.575"
                  stop-color="#4E56EA"
                />
              </lineargradient>
              <clippath
                id="clip0_681_72"
              >
                <rect
                  fill="white"
                  height="28"
                  transform="translate(6 6)"
                  width="28"
                />
              </clippath>
            </defs>
          </svg>
        </div>
        <div
          class="mocked-styled-15 el-menu-item-text-wrapper"
        >
          <div
            class="mocked-styled-16 el-app-switcher-menu-item-name"
          >
            Reapit Sales
          </div>
          <div
            class="mocked-styled-17 el-app-switcher-menu-item-description"
          >
            Real Estate CRM
          </div>
        </div>
      </a>
      <a
        class="mocked-styled-13 el-app-switcher-menu-item-anchor"
        href="https://www.test.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="mocked-styled-14 el-app-switcher-menu-item-logo"
        >
          <svg
            fill="none"
            font-size="40"
            height="1em"
            viewBox="0 0 40 40"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 8C0 3.58172 3.58172 0 8 0H32C36.4183 0 40 3.58172 40 8V32C40 36.4183 36.4183 40 32 40H8C3.58172 40 0 36.4183 0 32V8Z"
              fill="#ECF3FF"
            />
            <g
              clip-path="url(#clip0_681_8218)"
            >
              <path
                d="M30.9103 23.9576L26.3855 19.4328L30.6821 15.1362C32.4566 13.3617 32.4566 10.4546 30.6821 8.68007C28.9076 6.90557 26.0005 6.90557 24.226 8.68007L16.8865 16.1246L16.8718 16.1519V11.7748C16.8718 9.26317 14.818 7.20935 12.3064 7.20935C9.79477 7.20935 7.74097 9.26317 7.74097 11.7748V27.0501C7.74097 28.3052 8.25337 29.4463 9.08147 30.2737L9.07237 30.2898C10.8469 32.0643 13.754 32.0643 15.5285 30.2898L19.9581 25.8601L24.4542 30.4129C26.2287 32.1874 29.1358 32.1874 30.9103 30.4129C32.6848 28.6384 32.6848 25.7313 30.9103 23.9569V23.9576ZM12.3064 7.89817C14.4232 7.89817 16.1403 9.61528 16.1403 11.7321C16.1403 13.8489 14.4232 15.566 12.3064 15.566C10.1896 15.566 8.47247 13.8489 8.47247 11.7321C8.47247 9.61528 10.1896 7.89817 12.3064 7.89817ZM12.3337 30.8848C10.2309 30.8848 8.52497 29.181 8.52497 27.0761C8.52497 24.9712 10.2288 23.2674 12.3337 23.2674C14.4386 23.2674 16.1424 24.9712 16.1424 27.0761C16.1424 29.181 14.4386 30.8848 12.3337 30.8848ZM23.6107 11.564C23.7612 9.45215 25.5952 7.86245 27.7071 8.01295C29.819 8.16345 31.4087 9.99745 31.2582 12.1094C31.1077 14.2206 29.2737 15.8109 27.1618 15.6604C25.0499 15.5099 23.4602 13.6759 23.6107 11.564ZM27.9129 31.1221C25.7471 31.2432 23.8907 29.5863 23.7696 27.4184C23.6485 25.2526 25.3054 23.3962 27.4733 23.2751C29.6412 23.154 31.4955 24.8109 31.6166 26.9788C31.7377 29.1467 30.0808 31.001 27.9129 31.1221Z"
                fill="#4E56EA"
              />
            </g>
            <defs>
              <clippath
                id="clip0_681_8218"
              >
                <rect
                  fill="white"
                  height="28"
                  transform="translate(6 6)"
                  width="28"
                />
              </clippath>
            </defs>
          </svg>
        </div>
        <div
          class="mocked-styled-15 el-menu-item-text-wrapper"
        >
          <div
            class="mocked-styled-16 el-app-switcher-menu-item-name"
          >
            KeyWhere
          </div>
          <div
            class="mocked-styled-17 el-app-switcher-menu-item-description"
          >
            Real-time Key Management
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
